File: frontend-api/src/main/java/no/nav/melosys/tjenester/gui/dto/VirksomheterDto.java
package no.nav.melosys.tjenester.gui.dto;

import static no.nav.melosys.domain.kodeverk.Avklartefaktatyper.VIRKSOMHET;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

import no.nav.melosys.service.avklartefakta.AvklartefaktaDto;

public class VirksomheterDto {

    private List<String> virksomhetIDer;

    public List<String> getVirksomhetIDer() {
        return virksomhetIDer;
    }

    public void setVirksomhetIDer(List<String> virksomhetIDer) {
        this.virksomhetIDer = virksomhetIDer;
    }

    public static VirksomheterDto av(Set<AvklartefaktaDto> avklartefaktas) {
        List<String> virksomheter = new ArrayList<>();
        avklartefaktas.stream()
            .filter(avklartefakta -> VIRKSOMHET.getKode().equals(avklartefakta.getReferanse()) && VIRKSOMHET.equals(avklartefakta.getAvklartefaktaType()))
            .forEach(avklartefakta -> virksomheter.add(avklartefakta.getSubjektID()));

        VirksomheterDto virksomheterDto = new VirksomheterDto();
        virksomheterDto.setVirksomhetIDer(virksomheter);
        return virksomheterDto;
    }
}

File: frontend-api/src/main/kotlin/no/nav/melosys/tjenester/gui/dto/oppsummertefakta/VirksomheterDto.kt
package no.nav.melosys.tjenester.gui.dto.oppsummertefakta

import no.nav.melosys.domain.kodeverk.Avklartefaktatyper
import no.nav.melosys.service.avklartefakta.AvklartefaktaDto

class VirksomheterDto {
    @JvmField
    var virksomhetIDer: List<String>? = null

    companion object {
        fun av(avklartefaktas: Set<AvklartefaktaDto>): VirksomheterDto {
            val virksomheter: MutableList<String> = ArrayList()
            avklartefaktas.stream()
                .filter { avklartefakta: AvklartefaktaDto -> Avklartefaktatyper.VIRKSOMHET.kode == avklartefakta.referanse && Avklartefaktatyper.VIRKSOMHET == avklartefakta.avklartefaktaType }
                .forEach { avklartefakta: AvklartefaktaDto -> virksomheter.add(avklartefakta.subjektID) }

            val virksomheterDto = VirksomheterDto()
            virksomheterDto.virksomhetIDer = virksomheter
            return virksomheterDto
        }
    }
}
